src/stable_hasher.rs

```diff
@@ -60,23 +110,36 @@ pub struct StableHasher {
 /// }
 /// ```
 pub trait StableHasherResult: Sized {
+    type Hash;
+
     /// Retrieving the finalized state of the [`StableHasher`] and construct
     /// an [`Self`] containing the hash.
-    fn finish(hasher: [u64; 2]) -> Self;
+    fn finish(hash: Self::Hash) -> Self;
 }
 
-impl StableHasher {
+impl<H: ExtendedHasher + Default> StableHasher<H> {
     /// Creates a new [`StableHasher`].
     ///
     /// To be used with the [`Hasher`] implementation and [`StableHasher::finish`].
+    // TODO: Should this exist as is? or maybe it should take an `ExtendedHasher` as input?
```

Comment:
Taking an `ExtendedHasher` seems like a good idea.

Reply:
I ended-up adding a new function `StableHasher::with_hasher` for taking an `ExtendedHasher` as input. I didn't wanted the `new` function to have some caveats (should not be used, otherwise it's not stable). I added those caveats to the new function and kept new as a straight forward alias to default.